### DISCUSSION TITLE
Be able to verify a docker images as a publisher

### DISCUSSION THREAD

**Discussion by pjgg** [quay/quay#1274] [Ideas]
General idea: `Docker Verified Publishers provide developers with validation that these images come from trusted sources, reducing the risk of pulling malicious images from the impostor repositories.`

As a developer, I would like to be able to push an image plus some signatures(or something else) that allow me to verify those images. 

Motivation: https://github.com/bitnami-labs/sealed-secrets/issues/822#issuecomment-1104116084

Other implementations: https://docs.docker.com/docker-hub/publish/

**Comment by pjgg** [comment]
discursion open on the other side: https://github.com/bitnami/bitnami-docker-postgresql/issues/336

**Comment by flavianmissi** [comment]
AFAIK docker verified publishers is really just about image signing, which quay (and quay.io) fully supports via [cosign](https://access.redhat.com/documentation/en-us/red_hat_quay/3/html/use_red_hat_quay/oci-intro#cosign-oci-with-quay).

**Comment by dmesser** [comment]
Supporting storing image signatures is possible on Quay.io today, but for a verification mechanism a bit more is involved. We'd basically need to integrate with the SigStore.dev infrastructure to tie back those signatures to signer identities that you trust. This will take time for vendors and developers to adopt but we'll get there eventually.

Until then, publisher verification mechanism would probably require bespoke processes and resources to be effective. Quay.io would need to be trusted middle man in order to give assurances to user that quay.io/bitnami is indeed provided by Bitnami and not by an impostor. No small feat.

**Reply by kroontyc** [reply]
lol
